prime-router/src/main/kotlin/fhirengine/translation/HL7toFhirTranslator.kt

```diff
@@ -24,7 +24,29 @@ class HL7toFhirTranslator(
     private val configFolderPath: String = "./metadata/HL7/catchall",
     private val messageEngine: HL7MessageEngine = FhirTranscoder.getMessageEngine(),
 ) : Logging {
+
+    private val templates: Map<String, HL7MessageModel> =
+        ResourceReader(ConverterConfiguration(configFolderPath)).messageTemplates
+
     companion object {
+
+        // don't load the templtaes every time
+        // get individual isntances of the transaltor
+        // translro has its own templates
+
+        private val hl7ToFhirTranslatorInstances =
+            Collections.synchronizedMap(mutableMapOf<String, HL7toFhirTranslator>())
+
+        fun getHL7ToFhirTranslatorInstance(configFolderPath: String = "./metadata/HL7/catchall"): HL7toFhirTranslator {
+            return hl7ToFhirTranslatorInstances.getOrPut(configFolderPath) {
+                HL7toFhirTranslator(configFolderPath)
+            }
+        }
+
+        /**
+         * TODO tech debt ticket to clean everything below
```

Comment:
create ticket and link

Reply:
Actually, the clean up here was very easy so doing it as part of this PR.